### PR TITLE
Enhancement: Next releases controller

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

**Comment by core23** [comment]
This PR looks interesting :rocket: 

What's the goal of it? Will we able to create a new release via a web page?

File: src/Controller/DefaultController.php [hunk 1]
```diff
@@ -33,4 +35,29 @@ public function index(): Response
             ]
         );
     }
+
+    /**
+     * @Route("/healt", name="health")
```
Comment (core23) [inline src/Controller/DefaultController.php]:
```suggestion
     * @Route("/health", name="health")
```

**Comment by OskarStark** [comment]
> What's the goal of it? Will we able to create a new release via a web page?

For now, I just want to create an overview and a list of PR's which does not have a stability label and/or a changelog
CleanShot 2020-09-30 at 16 38 00
CleanShot 2020-09-30 at 16 38 11
CleanShot 2020-09-30 at 16 39 14

File: src/Domain/Value/NextRelease.php [hunk 2]
```diff
@@ -160,6 +160,14 @@ public function isNeeded(): bool
         return $this->nextTag->toString() !== $this->currentTag->toString();
     }
 
+    public function canBeReleased(): bool
```
Comment (VincentLanglet) [inline src/Domain/Value/NextRelease.php:163]: A PR like https://github.com/sonata-project/SonataAdminBundle/pull/6397 is needed before a release
- Updating the changelog
- Changing the 3.x occurences

The two solutions is see would be
- Make this PR automatic (but how ?)
- Using a label for the pre-release PR

Do you have any other idea ?

File: src/Controller/NextReleaseForProjectController.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        } catch (UnknownProject $e) {
+            throw new NotFoundHttpException($e->getMessage());
+        }
+
+        $release = $this->determineNextRelease->__invoke($project);
+
+        $content = $this->twig->render(
+            'releases/project.html.twig',
+            [
+                'release' => $release,
```
Comment (localheinz) [inline src/Controller/NextReleaseForProjectController.php:54]:
```diff
diff --git a/src/Controller/NextReleaseOverviewController.php b/src/Controller/NextReleaseOverviewController.php
index 488ae46..e3d9336 100644
--- a/src/Controller/NextReleaseOverviewController.php
+++ b/src/Controller/NextReleaseOverviewController.php
@@ -17,6 +17,7 @@ use App\Action\DetermineNextRelease;
 use App\Action\Exception\CannotDetermineNextRelease;
 use App\Action\Exception\NoPullRequestsMergedSinceLastRelease;
 use App\Config\Projects;
+use App\Domain\Value\NextRelease;
 use App\Domain\Value\Project;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
@@ -52,10 +53,14 @@ final class NextReleaseOverviewController
             return $releases;
         }, []);

+        $releasesSortedByNumberOfPullRequests = usort($releases, static function (NextRelease $a, NextRelease $b): int {
+            return count($b->pullRequests()) <=> count($a->pullRequests());
+        });
+
         $content = $this->twig->render(
             'releases/overview.html.twig',
             [
-                'releases' => $releases,
+                'releases' => $releasesSortedByNumberOfPullRequests,
             ]
         );
```

?

File: src/Controller/NextReleaseOverviewController.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+            $releases[] = $release;
+
+            return $releases;
+        }, []);
+
+        $content = $this->twig->render(
+            'releases/overview.html.twig',
+            [
+                'releases' => $releases,
```
Comment (localheinz) [inline src/Controller/NextReleaseOverviewController.php:58]:
```diff
diff --git a/src/Controller/NextReleaseOverviewController.php b/src/Controller/NextReleaseOverviewController.php
index 488ae46..e3d9336 100644
--- a/src/Controller/NextReleaseOverviewController.php
+++ b/src/Controller/NextReleaseOverviewController.php
@@ -17,6 +17,7 @@ use App\Action\DetermineNextRelease;
 use App\Action\Exception\CannotDetermineNextRelease;
 use App\Action\Exception\NoPullRequestsMergedSinceLastRelease;
 use App\Config\Projects;
+use App\Domain\Value\NextRelease;
 use App\Domain\Value\Project;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
@@ -52,10 +53,14 @@ final class NextReleaseOverviewController
             return $releases;
         }, []);

+        $releasesSortedByNumberOfPullRequests = usort($releases, static function (NextRelease $a, NextRelease $b): int {
+            return count($b->pullRequests()) <=> count($a->pullRequests());
+        });
+
         $content = $this->twig->render(
             'releases/overview.html.twig',
             [
-                'releases' => $releases,
+                'releases' => $releasesSortedByNumberOfPullRequests,
             ]
         );
```

?

**Comment by OskarStark** [comment]
It is working, I merge this

**Pull request merged**